shared/src/hooks/useFetchAssetBalanceData.tsx

```diff
@@ -42,19 +42,21 @@ const useFetchAssetBalanceData = (
       console.time("Asset Balance Data Fetch");
     }
 
-    if (!active) {
+    if (!active || !chainId) {
       setData({ ...defaultUserAssetBalanceData, loading: false });
       return;
     }
 
     const responses = await Promise.all(
       AssetsList.map(async (asset) => {
-        const balance = await (asset === "WETH" || asset === 'WAVAX'
+        const token = getERC20Token(library, asset.toLowerCase() as ERC20Token, chainId);
+        if (!token) {
+          return { asset, balance: undefined };
+        }
+
+        const balance = await (asset === "WETH" || asset === "WAVAX"
```

Comment:
Could be good to refactor this to a helper function `isNativeToken(asset)`

Reply:
Fixed!